File: fabric/src/main/java/com/selfdot/cobblemontrainers/fabric/server/CobblemonTrainersFabricServer.java
package com.selfdot.cobblemontrainers.fabric.server;

import com.selfdot.cobblemontrainers.CobblemonTrainers;
import com.selfdot.cobblemontrainers.fabric.FabricPermissionValidator;
import net.fabricmc.api.DedicatedServerModInitializer;

public class CobblemonTrainersFabricServer implements DedicatedServerModInitializer {

    @Override
    public void onInitializeServer() {
        CobblemonTrainers.INSTANCE.setPermissionValidator_(new FabricPermissionValidator());
    }

}

File: fabric/src/main/kotlin/com/selfdot/cobblemontrainers/fabric/server/CobblemonTrainersFabricServer.kt
package com.selfdot.cobblemontrainers.fabric.server

import com.selfdot.cobblemontrainers.CobblemonTrainers
import com.selfdot.cobblemontrainers.fabric.FabricPermissionValidator
import net.fabricmc.api.DedicatedServerModInitializer

class CobblemonTrainersFabricServer : DedicatedServerModInitializer {

    override fun onInitializeServer() {
        CobblemonTrainers.permissionValidator = FabricPermissionValidator()
    }

}
